### PR TITLE
Remove spreads in Addon components

### LINKED ISSUE #1416: Don't spread add-on and installation in mapStateToProps
`InstallButton` and some add-on components depend on properties of `addon` and `installation` to be set at the root rather than being namespaced. This means the status of an add-on and the status of its installation can collide and cause annoying errors where `status` is expected to be an installation status but is actually an add-on status.

We should return `{ addon, installation }` from mapStateToProps instead of `{ ...addon, ...installation }`.

### PR DESCRIPTION
**Pull request by willdurand**
Fix #1416

---

- [x] remove spreads in `Addon` (amo)
- [x] remove spreads in `Addon` (disco)

### CODE REVIEW

File: src/disco/components/Addon/index.js [hunk 1]
```diff
@@ -327,16 +335,13 @@ export function mapStateToProps(state, ownProps) {
   // `ownProps.guid` is already "normalized" with `getGuid()` in the
   // `DiscoPane` container component.
   const installation = state.installations[ownProps.guid] || {};
-  const addon = getAddonByGUID(state, ownProps.guid);
```
Comment (willdurand) [inline src/disco/components/Addon/index.js]: This is not needed because `DiscoPane` passes the `addon` object to each `Addon` component.

**Comment by kumar303** [comment]
Could you take a look at the conflicts?

File: tests/unit/amo/components/TestAddon.js [hunk 2]
```diff
@@ -1623,17 +1623,6 @@ describe(__filename, () => {
...
-      const description = 'whatever';
-      fetchAddon({ addon: { ...fakeAddon, description } });
-      const props = _mapStateToProps();
-
-      // Make sure a random addon prop gets passed as a component prop
-      // so that the withInstallHelpers HOC works.
-      expect(props.description).toEqual(description);
-    });
-
     it('must convert all installed addon props to component props', () => {
```
Comment (bobsilverberg) [inline tests/unit/amo/components/TestAddon.js]: This doesn't seem like a valid description for this test case anymore.

File: tests/unit/amo/components/TestAddon.js [hunk 3]
```diff
@@ -1644,11 +1633,11 @@ describe(__filename, () => {
           status: INSTALLED,
         }),
       );
-      const { needsRestart } = _mapStateToProps();
+      const { installStatus } = _mapStateToProps();
 
       // Make sure a random installedAddon prop gets passed as a component prop
       // so that the withInstallHelpers HOC works.
-      expect(needsRestart).toEqual(false);
+      expect(installStatus).toEqual(INSTALLED);
```
Comment (bobsilverberg) [inline tests/unit/amo/components/TestAddon.js]: Nit: set a `const` for `INSTALLED` at the top of the test.

File: tests/unit/amo/components/TestAddon.js (hunk 2 above)
Comment (willdurand) [reply tests/unit/amo/components/TestAddon.js]: oh, good catch!

**Pull request merged**